packages/xforms-engine/src/client/form/FormInstance.ts

```diff
@@ -33,9 +33,61 @@ export type FormInstanceCreateMode = 'create';
  */
 export type FormInstanceRestoreMode = 'restore';
 
+/**
+ * Represents an instance which:
+ *
+ * - was previously serialized
+ * - was **submitted**, often referred to as a "submission" (noun)
+ * - is being initialized to produce an edited instance of that "submission"
+ *
+ * An edited instance may be populated from one or more network-accessible
+ * resources—its serialized instance XML, and any _instance attachments_—which
+ * are resolved in a manner similar to that of a form definition and its _form
+ * attachments_.
+ *
+ * Once resolved, an edited instance is restored from its previously filled
+ * state _as it was submitted_, consistent with
+ * {@link FormInstanceRestoreMode | restoring an instance}, with the following
+ * exceptions:
+ *
+ * - If the form or instance does not define an explicit
+ *   {@link https://getodk.github.io/xforms-spec/#metadata | `deprecatedID` metadata element},
+ *   one will be created.
+ *
+ * - If the instance has an `instanceID` metadata element, that element's value
+ *   will be assigned to the instance's `deprecatedID`.
+ *
+ * - If applicable, a new instance id will be assigned:
+ *
+ *     - If the instance has an `instanceID` metadata element, and the form
+ *       explicitly defines a
+ *       {@link https://getodk.github.io/xforms-spec/#preload-attributes | `uid` preload attribute}
+ *       for its model binding, a new instance id value will be assigned to that
+ *       element _as if it were the
+ *       {@link https://getodk.github.io/xforms-spec/#event:odk-instance-first-load | instance's first load}_.
+ *
+ *     - **Consistent with {@link FormInstanceCreateMode | instance creation}:**
+ *       if the form **DOES NOT** define an `instanceID` element or model
+ *       binding, the `uid` preload behavior described above is **implied** as
+ *       the form's default behavior for that metadata; in which case whether
+ *       the instance defines an `instanceID` element explicitly or not, it will
+ *       be treated as if it had, behaving as described in the point above.
+ *
+ * **NOTE:** the behavior described for an instance's `instanceID` metadata
+ * element and a form's `uid` preload attribute are a special case applying only
+ * to that element and that preload attribute. In other words, when initializing
+ * instance state for editing:
+ *
+ * - The `odk-instance-first-load` event itself will **NOT** be triggered
+ *
+ * - Preload attributes _other than `uid`_ will **NOT** be recomputed
+ */
```

Comment:
@lognaturel This describes both the I/O aspects of this PR, as well as semantics I'm implementing in #352. If you get a chance, could you take a quick look and correct anything I might have gotten wrong about the semantics of the latter?

Reply:
Since the PR has now been approved, and in the interest of moving forward on #352, I've revised this JSDoc to be less specific about application of the `preload="uid"` case (matching the current behavior there, pending tests to validate).